activestorage/app/models/active_storage/blob.rb

```diff
@@ -116,6 +116,12 @@ def create_before_direct_upload!(key: nil, filename:, byte_size:, checksum:, con
     def generate_unique_secure_token(length: MINIMUM_TOKEN_LENGTH)
       SecureRandom.base36(length)
     end
+
+    def generate_unique_secure_key(key: nil, minimum_token_length: MINIMUM_TOKEN_LENGTH)
```

Comment:
This method should be implemented at instance level in order to allow interpolation mechanism with local attributes.

I suggest to either choose a different name for the option (for example `key_prefix`, because it actually is appending something) or to make it replacing actual value of `key` (to avoid confusion)